routes/delete-file.js

```diff
@@ -3,29 +3,25 @@ var utils = require("./utils");
 
 module.exports = function(config) {
   return function(req, res) {
-    if(!req.body || !req.body.path || !req.body.dateUpdated) {
-      res.status(400).send({error: "Request body missing data"});
-      return;
-    }
-
-    var path = utils.stripProjectRoot(req.session.project.root, req.body.path);
+    var path = req.body.path;
     var user = req.user;
     var project = req.session.project.meta;
-    var existingFile = utils.getFileFromArray(req.session.project.files, path);
+    var fileId = req.params.fileId;
 
-    if(!existingFile) {
+    if(!fileId) {
       res.status(400).send({error: "No file representation found for " + path});
```

Comment:
This isn't the correct error message anymore IMO, since it means a piece of data was omitted.